runner_service/utils.py

```diff
@@ -206,9 +207,25 @@ def timeout_handler():
             raise SSHTimeout
 
         socket.setdefaulttimeout(self.timeout)
-        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        socket.getaddrinfo(self.host, self.port, 0, 0, socket.SOL_TCP)
```

Comment:
I think you didn't understood my previous comment - if you use the return value of the `socket.getaddrinfo` you can actually get rid of all the if-ology below. Also, it's probably better to use `socket.socket` as a context manager. Smth like this should do:

```
try:
    family, _, _, _, addr = socket.getaddrinfo(self.host, self.port, 0, socket.SOCK_STREAM)
except socket.gaierror:
    raise HostNotFound

with socket.socket(family, socket.SOCK_STREAM) as s:
    try:    
        s.connect(addr)
    except ConnectionRefusedError:
        raise SSHNotAccessible
    except socket.timeout:
        raise SSHTimeout
```